### ISSUE TITLE
How to build DOS target?

### ISSUE THREAD

**Issue by fsb4000** [LADSoft/OrangeC#315]
Sorry for noob question, but how I can build a DOS binary?

I tried:
```
occ /1  /I"include" /I"3rd" -c -o src/main.o src/main.c
occ -o bin/solution.exe src/main.o
```
The program works, but it is  a Windows program.

I read: 
>The existing backend code generation programs support WIN32 and MSDOS executable formats

How can I change formats to MSDOS?
Thank you in advance!

**Comment by fsb4000** [comment]
also I found https://orangec.readthedocs.io/en/latest/general/DLLE/
but I could not figure it out :(

**Comment by fsb4000** [comment]
Oh, sorry again.
How can I enable FPU or something?
I'm trying build some simple example with complex numbers:
https://en.cppreference.com/w/c/numeric/complex/creal
```c
#include <stdio.h>
#include <complex.h>
 
int main(void)
{    
    double complex z = 1.0 + 2.0*I;
    printf("%f%+fi\n", creal(z), cimag(z));
}
```
But I got exceptions :(

**Comment by chuggafan** [comment]
What build? Because OCC recently dropped DOS support entirely as it uses SSE instructions, you have better luck with cc386.

**Comment by LADSoft** [comment]
yeah i'm always wanting to get rid of DOS anyway lol!   But as it turns out SSE2 under dos isn't much of an issue, at least I don't think it is...  I'm pretty sure it would be possible to get FP math working again on DOS even with the current OCC design and it wouldn't even take much effort.    The more pertinent issue is that if you have DOS you probably don't have a CPU capable of SSE2 though sigh.

**Comment by GitMensch** [comment]
Can we please reopen this?

If DOS is actually planned to be removed (I'd personally prefer to not limit to the OS but only to the CPU, as discussed before) then `/T:DOS32` should raise an error ("use version N.N.N or CC386") and sources (compilation) for cldos.l should be removed.
Otherwise (preferably) the codegen should be fixed.

I've actually planned to go on testing for a working cross-compilation once the GnuCOBOL testbed passes...

**Comment by fsb4000** [comment]
> The more pertinent issue is that if you have DOS you probably don't have a CPU capable of SSE2 though sigh

Yes, you are right:( 
DosBox support only 386, 486 and Pentium CPU emulation...
 >cputype: CPU Type used in emulation. auto is the fastest choice.
 >Possible values: auto, 386, 386_slow, 486_slow, pentium_slow, 386_prefetch.

**Comment by GitMensch** [comment]
@fsb4000: I suggest using VirtualBox or a different VPC solution and FreeDos.

**Comment by LADSoft** [comment]
ok i'll fix it within a few days.   If virtualbox isn't up to it I can always power up an old DOS box lol!

**Comment by fsb4000** [comment]
@GitMensch sure, I have FreeDos.
By the way if you have an opcode error while installation, you should press 'Tab' and add 'raw'.
Images:
![freedos1](https://i.imgur.com/crgG7o0.png)
![freedos2](https://i.imgur.com/VyCy0AD.png)

**Comment by LADSoft** [comment]
hm, wonder what is up with that?

Actually the compiler is supposed to run on DOS, using japheth's HXDOS environment.   It can generate executables for HXDOS among other things...   there is an installer...   by default under DOS it uses DOS32A as the extender for programs it generates.    That can be forced on windows by setting the ORANGECDOS environment variable...   There are some problems with all that though, guess I let it go too long without regression testing.    I'll get it patched up soonest...

**Comment by LADSoft** [comment]
I fixed this and made a release '6.0.40'
